### PR TITLE
Remove special case deferred methods

### PR DESCRIPTION
**Pull request by orangejenny**
https://jquery.com/upgrade-guide/3.0/#breaking-change-special-case-deferred-methods-removed-from-jquery-ajax

@esoergel / @czue

### CODE REVIEW

File: corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js [hunk 1]
```diff
@@ -27,12 +27,13 @@ var NewStripeCard = function(data){
...
-        }).success(function(data){
-            $("#card-modal").modal('hide');
-            $("#success-modal").modal('show');
-            cardManager.wrap(data);
-            self.reset();
+            data: self.unwrap(),
+            success: function(data) {
+                $("#card-modal").modal('hide');
+                $("#success-modal").modal('show');
+                cardManager.wrap(data);
```
Comment (dimagimon) [inline corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js]: 'cardManager' is not defined. (no-undef)

File: corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js [hunk 2]
```diff
@@ -104,11 +105,12 @@ var StripeCard = function(card, baseUrl){
...
         $.ajax({
             type: "DELETE",
-            url: self.url
-        }).success(function(data){
-            cardManager.wrap(data);
-            $(button.currentTarget).closest(".modal").modal('hide');
-            $("#success-modal").modal('show');
+            url: self.url,
+            success: function(data) {
+                cardManager.wrap(data);
```
Comment (dimagimon) [inline corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js]: 'cardManager' is not defined. (no-undef)

File: corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js [hunk 3]
```diff
@@ -124,9 +126,10 @@ var StripeCard = function(card, baseUrl){
         return $.ajax({
             type: "POST",
             url: self.url,
-            data: data
-        }).success(function(data){
-            cardManager.wrap(data);
+            data: data,
+            success: function(data) {
+                cardManager.wrap(data);
```
Comment (dimagimon) [inline corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js]: 'cardManager' is not defined. (no-undef)

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 4]
```diff
@@ -63,21 +63,25 @@ var ExportManager = function (o) {
...
-                        autoRefresh = false;
+                    $.get({
+                        url: params.data.download_url,
+                        success: function(data) {
+                            self.$modal.find(self.exportModalLoadedData).html(data);
+                            self.setUpEventTracking({
+                                xmlns: params.xmlns,
+                                isBulkDownload: params.isBulkDownload,
+                                exportName: params.exportName,
+                                isMultimedia: params.isMultimedia
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Missing trailing comma. (comma-dangle)

**Comment by orangejenny** [comment]
ignoring dimagimon; this PR just moves code around

**Review by esoergel — APPROVED** [review]
Looks good.  Ignoring whitespace makes it easier to review: https://github.com/dimagi/commcare-hq/pull/13248/files?w=1

**Comment by orangejenny** [comment]
approved & tests pass => merging

**Pull request merged**